lib/cdo/cloud_formation/lambda.rb

```diff
@@ -34,6 +34,61 @@ def js_erb(filename, **args)
       end
     end
 
+    # Prepare (package) a CloudFormation custom Resource Lambda that uses node runtime to be deployed by CloudFormation
+    # by installing dependencies, zipping the Lambda directory, uploading to S3, and returning the S3 URI to
+    # populate the `AWS::Lambda::Function` `Code` Property.
+    # Assumes CloudFormation custom Resource Lambdas are in `/aws/cloudformation/lambdas/`.
+    def package_node_lambda(relative_directory)
+      install_node_dependencies(relative_directory)
+      return zip_directory(relative_directory)
+    end
+
+    # Install npm packages used by a lambda to prepare the directory the lambda is in for being zipped and uploaded.
+    # Assumes CloudFormation custom Resource Lambdas are in `/aws/cloudformation/lambdas/`.
+    def install_node_dependencies(relative_directory)
+      absolute_directory = aws_dir('cloudformation/lambdas' + '//' + relative_directory)
+      Dir.chdir(absolute_directory) do
+        # Use the `ci` parameter to only install the versions identified in the lock file.
+        # Use `--only=prod` to skip dev dependencies.
+        RakeUtils.npm_install 'ci --only=prod'
+      end
+    end
+
+    # Zip a directory containing a Lambda's source code and dependencies, upload to S3, and return the S3 location
+    # to assist in populating the `Code` Property of a CloudFormation template `AWS::Lambda::Function` Resource.
+    # Assumes CloudFormation custom Resource Lambdas are in `/aws/cloudformation/lambdas/`.
```

Comment:
Is this limited to Cloudformation Custom Resources, or would this be fine for any lambda?

Reply:
This is for any Lambda that we need to deploy with our existing rake `stack` tasks (using our `Cdo::CloudFormation::Lambda` module). My hope is that in the long term we would use more modern methods (like we do in javabuilder) to package and deploy lambdas.

Reply:
We could remove the 'custom resource' bit then, as this could be used for any lambda.

Reply:
Ah. Good point. I’ll update the comments to note that.

Reply:
Good point. Done!